R/R/sqlPackage.R

```diff
@@ -1267,11 +1268,24 @@ downloadDependentPackages <- function(pkgs, destdir, binaryPackages, sourcePacka
 
         if (length(downloadedPkg) < 1)
         {
+            write(sprintf("%s  Could not find binary version in repo, trying source instead...", pkgTime()), stdout())
+
             #
             # try source package if binary package isn't there
-            #
-            downloadedPkg <- utils::download.packages(pkg$Package, destdir = destdir,
-                                                      available = sourcePackages, type = "source", quiet = TRUE)
+
+            if(serverVersion$sysname == Sys.info()[['sysname']])
+            {
+                #
+                # If the server and client are the same type,
+                # with source packages, we may need to build the binary on the client
+                #
+                downloadedPkg = buildSourcePackage(pkg$Package, destdir, sourcePackages)
+            }
+            else
+            {
+                downloadedPkg <- utils::download.packages(pkg$Package, destdir = destdir,
+                                                          available = sourcePackages, type = pkgType, quiet = TRUE)

```

Comment:
You are correct about the case.

I think if it is cross plat and we only have source packages we should just try our current implementation, i.e. sending the source to the server and building there. it's possible that this is a box server and can modify permissions on the server machine, no need to fail without trying.

Also, not all source packages have the problem we ran into here. Some packages work fine because they don't need access to the restricted resources, so we may as well try imo.

Reply:
sure makes sense. we can try

Reply:
Can you add the comment in the code? Thanks.